runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -917,6 +919,41 @@ private <K1, K2, V> void groupByKeyAndSortValuesHelper(
           }
         });
 
+    registerTransformTranslator(
+        RedistributeByKey.class,
```

Comment:
@robertwb you might be the person to review DataflowRunner translation? For Reshuffle we don't have a translator but a more complex rewrites to a specialized GroupByKey. I opted to _not_ do that this time but translate more directly. I added ValidatesRunner tests for Redistribute that check parity with Reshuffle, at least in terms of that test suite.

CC @scwhittle

Reply:
Should we register a RedistributeArbitrary translator as well?  At least for streaming the entirely distinct bundles with entirely unique keys is rarely a good choice.  Or a TODO if out of scope for this change.

Reply:
I do want to do a custom translation, yea. I think the best way to do it is probably a black-box Dataflow primitive rather than client side. Filed b/330519228 internally.

Reply:
Is there a reason to not simply retain the existing composite (reshuffle) implementation (which can then be detected and overwritten by any runner?)

Reply:
I guess I was thinking in terms of portable runners. Here we have to have a special translator to set the `PropertyNames.ALLOW_DUPLICATES` property, right?

Reply:
Yes, that's right. (and Reshuffle doesn't have a translator because it just has an override with special windowing strategy as a flag value on GBK to tell v1 to reshuffle)